File: rust/src/models/capacity_fee_structure.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::billable_metric_id::BillableMetricId;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CapacityFeeStructure {
    pub metric_id: BillableMetricId,
}

impl CapacityFeeStructure {
    pub fn new(metric_id: BillableMetricId) -> Self {
        Self { metric_id }
    }
}

File: typescript/src/models/capacityFeeStructure.ts
// this file is @generated
import { type BillableMetricId, BillableMetricIdSerializer } from "./billableMetricId";

export interface CapacityFeeStructure {
  metricId: BillableMetricId;
}

export const CapacityFeeStructureSerializer = {
  _fromJsonObject(object: any): CapacityFeeStructure {
    return {
      metricId: BillableMetricIdSerializer._fromJsonObject(object["metric_id"]),
    };
  },

  _toJsonObject(self: CapacityFeeStructure): any {
    return {
      metric_id: BillableMetricIdSerializer._toJsonObject(self.metricId),
    };
  },
};
